### PR TITLE
:seedling: Unit tests for checks/evaluation/pinned_dependencies.go

### PR DESCRIPTION
**Pull request by naveensrinivasan**
Unit tests for checks/evaluation/pinned_dependencies.go

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [x] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: checks/evaluation/pinned_dependencies_test.go [hunk 1]
```diff
@@ -205,6 +207,34 @@ func Test_PinningDependencies(t *testing.T) {
 				NumberOfDebug: 0,
 			},
 		},
+		{
+			name: "various wanrings and info",
```
Comment (spencerschrock) [inline checks/evaluation/pinned_dependencies_test.go:211]: the purpose of the test isn't immediately clear to me from the name or what its expecting

File: checks/evaluation/pinned_dependencies_test.go [hunk 2]
```diff
@@ -354,3 +384,173 @@ func Test_maxScore(t *testing.T) {
...
+				},
+			},
+			expectedPinningResult: &worklowPinningResult{
+				thirdParties: 0,
+				gitHubOwned:  2,
+			},
+			expectedPinnedResult: map[checker.DependencyUseType]pinnedResult{},
+		},
+		{
+			name: "GitHub Action - GitHub-owned",
```
Comment (spencerschrock) [inline checks/evaluation/pinned_dependencies_test.go:530]: typo here. i assume the name should reflect third party

**Review by spencerschrock — APPROVED** [review]

File: checks/evaluation/pinned_dependencies_test.go (hunk 2 above)
Comment (naveensrinivasan) [reply checks/evaluation/pinned_dependencies_test.go]: 👍

File: checks/evaluation/pinned_dependencies_test.go (hunk 1 above)
Comment (naveensrinivasan) [reply checks/evaluation/pinned_dependencies_test.go]: Updated the name to reflect better message

**Pull request merged**